### ISSUE TITLE
Allow clients to set timezone on their profile

### ISSUE THREAD

**Issue by epichosts** [FOSSBilling/FOSSBilling#1028]
**Is your feature request related to a problem? Please describe.**
On tickets for example, clients can be confused at the time shown on a ticket and they think a response was made 3 hours earlier or later (3 is an example of course).

**Describe the solution you'd like**
The ability for clients to set their timezone on the profile so that times are accurate to their timezone when receiving/sending any correspondence through tickets, activation/suspensions and more.

**Describe alternatives you've considered**
The config.php file has the ability to set timezones across the whole setup but for obvious reasons this is not a suitable alternative.

**Additional context**
N/A

**Comment by John-S4** [comment]
I don't know what the best solution is here. Whether the timezone should be set automatically along with the locale and with the option to disable the auto selection and set manually, or if it should just be a profile field that needs to be set in the user profile.

**Comment by BelleNottelling** [comment]
I think the best bet would be to make it an option that can be selected by the client during sign-up. We can give them a list of timezones and then us JS to automatically select what is *probably* the right one & then they can change it if it's wrong.

Unlike the locale selection where it is important to get right before someone has created an account, the timezone being correct mostly only comes into play once they are a client with orders and tickets.

On top of that, trying to automatically determine the timezone server-side like we do the locale isn't really practical. We would need to hook into a 3rd party service, use a local database that's shipped with FOSSBilling to match IPs to their timezones, or find some way to get it via JS and send it to the backed. None of those are straightforward or foolproof methods so I'd rather avoid them if at all possible.

**Comment by John-S4** [comment]
> I think the best bet would be to make it an option that can be selected by the client during sign-up. We can give them a list of timezones and then use JS to automatically select what is _probably_ the right one & then they can change it if it's wrong.

Yes, that makes sense and sounds like a good solution. It should be editable later in the user settings as well as at signup.

**Comment by admdly** [comment]
I've added this in #3895. It would be great if you could check the latest preview (allow ~30mins for it to regenerate) or the next version, and advise if there are any issues so we can reopen this issue if needed.

**Comment by epichosts** [comment]
Hi @admdly, wow this is an old one! It looks like it's working however, as I can see the timezone configuration on both the admin and client side, and it seems to update the times accordingly on tickets etc on the client side when the timezone is adjusted. Thank you so much for your incredible work!
